base/shell.jl

```diff
@@ -256,19 +256,56 @@ shell_escape_posixly(args::AbstractString...) =
     shell_escape_wincmd(s::AbstractString)
     shell_escape_wincmd(io::IO, s::AbstractString)
 
-The unexported `shell_escape_wincmd` function escapes Windows
-`cmd.exe` shell meta characters. It escapes `()!^<>&|` by placing a
-`^` in front. An `@` is only escaped at the start of the string. Pairs
-of `"` characters and the strings they enclose are passed through
-unescaped. Any remaining `"` is escaped with `^` to ensure that the
-number of unescaped `"` characters in the result remains even.
-
-Since `cmd.exe` substitutes variable references (like `%USER%`)
-_before_ processing the escape characters `^` and `"`, this function
-makes no attempt to escape the percent sign (`%`).
-
-Input strings with ASCII control characters that cannot be escaped
-(NUL, CR, LF) will cause an `ArgumentError` exception.
+The unexported `shell_escape_wincmd` function escapes Windows `cmd.exe` shell
+meta characters. It escapes `()!^<>&|` by placing a `^` in front. An `@` is
+only escaped at the start of the string. Pairs of `"` characters and the
+strings they enclose are passed through unescaped. Any remaining `"` is escaped
+with `^` to ensure that the number of unescaped `"` characters in the result
+remains even.
+
+Since `cmd.exe` substitutes variable references (like `%USER%`) _before_
+processing the escape characters `^` and `"`, this function makes no attempt to
+escape the percent sign (`%`), the presence of `%` in the input may cause
+severe breakage, depending on where the result is used.
+
+Input strings with ASCII control characters that cannot be escaped (NUL, CR,
+LF) will cause an `ArgumentError` exception.
+
+The result is safe to pass as an argument to a command call being processed by
+`CMD.exe /S /C " ... "` (with surrounding double-quote pair) and will be
+received verbatim by the target application if the input does not contain `%`
+(else this function will fail with an ArgumentError). The presence of `%` in
+the input string may result in command injection vulnerabilities and may
+invalidate any claim of suitability of the output of this function for use as
+an argument to cmd (due to the ordering described above), so use caution when
+assembling a string from various sources.
+
+This function may be useful in concert with the `windows_verbatim` flag to
+[`Cmd`](@ref) when constructing process pipelines.
+
+Alternatively, the argument can be passed as an environment variable, which
+avoids the problem with `%` and the `windows_verbatim` flag:
+
+```julia
+cmdargs = shell_escape_wincmd("Passing args with %cmdargs% works 100%!")
+run(setenv(`cmd /C to_run %cmdargs%`, "cmdargs" => cmdargs))
```

Comment:
Maybe set it to `echo` so these run? 
```suggestion
run(setenv(`cmd /C echo %cmdargs%`, "cmdargs" => cmdargs))
```